### ISSUE TITLE
Add support for oauth access_token

### ISSUE THREAD

**Issue by cosigyn** [ValiantWind/cloudblox#21]
Requires an authorization header instead of x-api-key header.

`Authorization: Bearer <access_token>`

The rewrite should make it easier to implement this

**Comment by ValiantWind** [comment]
Thanks for the suggestion! I will implement this in the next release along with the rest of the rewrite. OAuth support is planned either with the completion of the rewrite or the release after.

**Comment by ValiantWind** [comment]
# TL;DR
v1.5.0 will include the finished with the breaking changes to the Open Cloud side of the package and OAuth support will be added in the following release (with Open Cloud Groups API support as well) to ease the transition needed for the breaking changes happening in v1.5.0.

A full explanation and list of the breaking changes will be included in the release, as well as a guide on how to migrate.

Expect the rewrite to be finished within a week or two, with OAuth support coming 5-7 days after.  

I know this looks like a lot of time, but I'm the only maintainer and making changes like this take a lot of time so I appreciate the patience 🙂 

***
# Long Explanation
I have just realized that implementing this would be more ideal if I implemented it with OAuth support entirely.  

There will be lots of breaking changes in the next release when the rewrite is finished, mostly with OpenCloud. 
I made the Web API rewrite very seamless so everyone would have to change as little code as possible. The same can't be said for the Open Cloud part of the package, as it was entirely rewritten and works differently yet in a more efficient and organized manner.

The release will include the finished rewrite with the breaking changes. I want to do this before implementing OAuth support so everyone can have time to transition before implementing more features in addition to the rewrite. This will make the process easy for both me and the users. My goal for the rewrite is to make the package more maintainable, expandable, and more efficient, while keeping the same functionality and simplicity. Adding OAuth in the same release as the rewrite would make things more complicated and take much longer than I want it to be.
